cookbook/persistance/key-value.md

```diff
@@ -0,0 +1,175 @@
+
+Setter and getter methods are available for all primitive types.
+
+## Supported types
+
+While is easy and convenient to use key-value storage, it has limitations:
+
+- Only primitive types can be used
+  - `int`, `double`, `bool`, `string` and `string list`
+- It's not designed to store a lot of data so it's not suitable as application cache.
+
+If we want to read more about Shared preferences on Android please visit [Shared preferences documentation](https://developer.android.com/guide/topics/data/data-storage.html#pref) on Android developers website.
+
+## Example
+
+```dart
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+void main() => runApp(new MyApp());
+
+class MyApp extends StatelessWidget {
+  // This widget is the root of our application.
+  @override
+  Widget build(BuildContext context) {
+    return new MaterialApp(
+      title: 'Shared preferences demo',
+      theme: new ThemeData(
+        primarySwatch: Colors.blue,
+      ),
+      home: new MyHomePage(title: 'Shared preferences demo'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  MyHomePage({Key key, this.title}) : super(key: key);
+  final String title;
+
+  @override
+  _MyHomePageState createState() => new _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  int _counter = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    _loadCounter();
+  }
+
+  //Loading counter value on start 
+  _loadCounter() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    setState(() {
+      _counter = (prefs.getInt('counter') ?? 0);
+    });
+  }
+  
+  //Incrementing counter after click
+  _incrementCounter() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    _counter = (prefs.getInt('counter') ?? 0) + 1;
+    setState(() {
+      _counter;
+    });
+    prefs.setInt('counter', _counter);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return new Scaffold(
+      appBar: new AppBar(
+        title: new Text(widget.title),
+      ),
+      body: new Center(
+        child: new Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: <Widget>[
+            new Text(
+              'You have pushed the button this many times:',
+            ),
+            new Text(
+              '$_counter',
+              style: Theme.of(context).textTheme.display1,
+            ),
+          ],
+        ),
+      ),
+      floatingActionButton: new FloatingActionButton(
+        onPressed: _incrementCounter,
+        tooltip: 'Increment',
+        child: new Icon(Icons.add),
+      ), // This trailing comma makes auto-formatting nicer for build methods.
+    );
+  }
+}
+
+```
+
+## Testing support
+
+We can populate SharedPreferences with initial values in our tests by running this code:
```

Comment:
For folks who are new, could you please describe where to place this code? e.g. in a `setupAll` method or the test file?

Reply:
I put simple note about this. I think in future, when test page will be written, we can change that to link to testing.